src/Sylius/Bundle/ShopBundle/Resources/views/Product/Show/_breadcrumb.html.twig

```diff
@@ -1,21 +1,23 @@
-<div class="ui breadcrumb">
-    <a href="{{ path('sylius_shop_homepage') }}" class="section">{{ 'sylius.ui.home'|trans }}</a>
-    <div class="divider"> / </div>
-    {% if product.mainTaxon is not null %}
-        {% set taxon = product.mainTaxon %}
-        {% set ancestors = taxon.ancestors|reverse %}
+{% from '@SyliusShop/Common/Macro/breadcrumb.html.twig' import breadcrumb %}
+{% set crumbs = [] %}
+{% if product.mainTaxon is not null %}
+    {% set taxon = product.mainTaxon %}
+    {% set ancestors = taxon.ancestors|reverse %}
 
-        {% for ancestor in ancestors %}
-            {% if ancestor.isRoot() %}
-                <div class="section">{{ ancestor.name }}</div>
-            {% else %}
-                <a href="{{ path('sylius_shop_product_index', {'slug': ancestor.slug, '_locale': ancestor.translation.locale}) }}" class="section">{{ ancestor.name }}</a>
-            {% endif %}
-            <div class="divider"> / </div>
-        {% endfor %}
-
-        <a href="{{ path('sylius_shop_product_index', {'slug': taxon.slug, '_locale': taxon.translation.locale}) }}" class="section">{{ taxon.name }}</a>
-        <div class="divider"> / </div>
-    {% endif %}
-    <div class="active section">{{ product.name }}</div>
-</div>
+    {% for ancestor in ancestors %}
+        {% if not ancestor.isRoot() %}
+            {% set crumbs = crumbs|merge([{
+                url: path('sylius_shop_product_index', {'slug': ancestor.slug, '_locale': ancestor.translation.locale}),
+                name: ancestor.name
+            }]) %}
+        {% endif %}
+    {% endfor %}
+    {% set crumbs = crumbs|merge([{
+        url: path('sylius_shop_product_index', {'slug': taxon.slug, '_locale': taxon.translation.locale}),
+        name: taxon.name
+    }]) %}
+{% endif %}
```

Comment:
IMHO this logic could be extracted to Twig function

Reply:
WDYT if it comes to something like this (take the idea, not the implementation itself):

```twig
{% set crumbs = getCrumbsFromTaxons(ancestors)|merge([{
        url: path('sylius_shop_product_index', {'slug': taxon.slug, '_locale': taxon.translation.locale}),
        name: taxon.name
    }]) %}
```

For me, having a twig function doing a merge is useless since the `merge` filter exists.
I understand for the loop on the taxons, it makes sens, but not for the second part.

Reply:
I was thinking about something like 
```twig
{% set crumbs = crumbTaxon(taxon)|merge([{
        url: path('sylius_shop_product_index', {'slug': taxon.slug, '_locale': taxon.translation.locale}),
        name: taxon.name
    }]) %}
```
Where it would return ready to use an array with URL and name. However, the main point here is, that this logic is duplicated and it is not just a merge, but it is a merge in a if statement in a for loop.

Reply:
Ok, I'll improve this.

Reply:
thank you :)